### PR TITLE
Fix Activation bootstrapping issues

### LINKED ISSUE #5055: Bootstrapping should properly support plugin activation
## User Story

As an admin I want to install GiveWP and have it work perfectly out-of-the-box.

## Details

When the bootstrapping was recently updated the activation process was broken due to the fact that bootstrapping in GiveWP works very differently for a plugin being activated versus a plugin running. This caused issues with not all of the setting defaults being set, and tables not being loaded. It also gave a false error that the database needed to be upgraded immediately after fresh installation.

## Expected Behavior

All default settings should be set and tables loaded when GiveWP is activated.

## Steps to Reproduce

1. Install GiveWP on a fresh WP instance
2. Activate GiveWP

## Acceptance Criteria

-   [x] GiveWP default settings are loaded when GiveWP is activated
-   [x] All GiveWP custom tables are created when GiveWP is activated

### PR DESCRIPTION
**Pull request by JasonTheAdams**
Resolves #5055 

## Description

This resolves the issue where activation was not working properly by doing the following:
1. Add a missing container binding to the `LegacyServiceProvider`
2. Run all the providers before installation to make sure everything is available
3. Add a require for a function being called too early in `Give_Session`

## Affects

The activation process for GiveWP

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions

Try installing and activating GiveWP on a fresh install of WordPress, as well as an existing installation.

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@ravinderk Please review this for me. It's a pretty simple fix.

File: src/ServiceProviders/LegacyServiceProvider.php [hunk 1]
```diff
@@ -24,7 +25,6 @@ public function register() {
 	 * @inheritDoc
 	 */
 	public function boot() {
```
Comment (kjohnson) [inline src/ServiceProviders/LegacyServiceProvider.php:27]: I know it needs to method to match the interface, but maybe add a comment like "this section intentionally left blank" or point to where the `register_activation_hook` was moved.

Not a blocker.

Comment (JasonTheAdams) [reply src/ServiceProviders/LegacyServiceProvider.php:27]: It's not really obvious that this is where the activation hook was registered previously. In the future, I'd like to add an abstract `ServiceProvider` class so we can remove the empty functions.

**Comment by JasonTheAdams** [comment]
@Benunc @DevinWalker This is approved, so please test this out. If it's working, then go ahead and merge this for release.

**Comment by JasonTheAdams** [comment]
Thanks for taking the time to make that image. It means a lot. 😁 😁

**Comment by DevinWalker** [comment]
@JasonTheAdams one thing I didn't think about before merging this is how are we going to remedy the users who already reported this and are missing tables currently? I believe what's needed here is an upgrade routine to help those folks get back to a proper db structure. Otherwise, the tables are going to remain uncreated, even at update.

**Comment by JasonTheAdams** [comment]
That's a good though, @DevinWalker. I believe I've seen that the database migrations want to get ran after the process fails, and I wonder if that actually resolves it. I've managed to get a fresh local site up and running before this fix was in place. The migrations would work post-activation, as the problem only occurs upon activation.

I'm curious if @ravinderk might have any insight on this.

**Comment by ravinderk** [comment]
@DevinWalker @JasonTheAdams As we know this is happening only for a fresh install, so we can ask the user to reinstall or reactivate `2.7.5` plugin because the last version on donation will not work at all  (session table missing).

**Pull request merged**